src/Component/RequestBuilder/FacetSet.php

```diff
@@ -108,28 +118,31 @@ public function buildComponent($component, $request)
      *
      * @param Request    $request
      * @param FacetField $facet
+     * @param compatible $compatible false if local parameters should be used
      */
-    public function addFacetField($request, $facet)
+    public function addFacetField($request, $facet, $compatibility_mode = true)
     {
         $field = $facet->getField();
 
+        $local_param = $compatibility_mode ? ['key' => $facet->getKey(), 'ex' => $facet->getExcludes()] : ['key' => $facet->getKey(), 'ex' => $facet->getExcludes(), 'facet.limit' => $facet->getLimit(), 'facet.sort' => $facet->getSort(), 'facet.prefix' => $facet->getPrefix(), 'facet.contains' => $facet->getContains(), 'facet.contains.ignoreCase' => $facet->getContainsIgnoreCase(), 'facet.offset' => $facet->getOffset(), 'facet.mincount' => $facet->getMinCount(), 'facet.missing' => $facet->getMissing(), 'facet.method' => $facet->getMethod()];
```

Comment:
The two long arrays should be formatted to set each param in a separate line. In your previous code your only mistake according to the style checker was that you didn'tplace the commas at the end of the line.